src/rpc/rpc_impl.rs

```diff
@@ -337,4 +341,26 @@ impl<N: Network, E: Environment> RpcFunctions<N> for RpcImpl<N, E> {
         }
         Ok(transaction.transaction_id())
     }
+
+    /// Returns the amount of shares submitted by a given prover.
+    async fn get_shares_for_prover(&self, prover: Value) -> Result<u64, RpcError> {
+        let prover: Address<N> = serde_json::from_value(prover)?;
+        Ok(self.operator.get_shares_for_prover(&prover))
+    }
+
+    /// Returns the amount of shares submitted to an operator in total.
+    async fn get_shares(&self) -> Result<u64, RpcError> {
+        let shares = self.operator.to_shares();
+        let mut res = 0;
+        for (_, share) in shares {
+            res += share.values().sum::<u64>();
+        }
+        Ok(res)
+    }
+
+    /// Returns a list of provers registered on an operator.
+    async fn get_provers(&self) -> Result<Value, RpcError> {
```

Comment:
```suggestion
    async fn get_provers(&self) -> Value {
```
same reasoning as with `get_shares`